src/clients/SpokePoolClient.ts

```diff
@@ -1064,4 +1075,79 @@ export class SpokePoolClient extends BaseAbstractClient {
 
     return deposit;
   }
+
+  async findDepositV3(depositId: number, destinationChainId: number, depositor: string): Promise<V3DepositWithBlock> {
+    // Binary search for block. This way we can get the blocks before and after the deposit with
+    // deposit ID = fill.depositId and use those blocks to optimize the search for that deposit.
+    // Stop searches after a maximum # of searches to limit number of eth_call requests. Make an
+    // eth_getLogs call on the remaining block range (i.e. the [low, high] remaining from the binary
+    // search) to find the target deposit ID.
+    //
+    // @dev Limiting between 5-10 searches empirically performs best when there are ~300,000 deposits
+    // for a spoke pool and we're looking for a deposit <5 days older than HEAD.
+    const searchBounds = await this._getBlockRangeForDepositId(
+      depositId,
+      this.deploymentBlock,
+      this.latestBlockSearched,
+      7
+    );
+
+    const tStart = Date.now();
+    const query = await paginatedEventQuery(
```

Comment:
I'm having trouble testing this in `relayer-v2` because this query doesn't seem to work with the Mock SpokePool client and we don't have the latest ABI's in relayer-v2 yet

Reply:
trying to test here but the `queryFilter` is returning nothing https://github.com/across-protocol/relayer-v2/pull/1155/files#diff-e73c5f246f260ad0f49e8990b9602cfb17f18b3e5f3c1627ec20e48db048c284R908

Reply:
This resolves down to Contract.filters.<event-name>, so the contract needs to be instantiated with the correct ABI. We're still instantiating with the old v2 ABIs, which is why it currently doesn't work if you're querying outside of the MockSpokePoolClient context (the mocked variant ignores these filters because it overrides the `SpokePoolClient._update()` method). We do have the option of instantiating a standalone contract with the new SpokePoolV3 ABI, but we still can't interact with it in hre so it wouldn't gain us much ground.

Thinking about this a bit more, it's arguably a layering violation that we sometimes query via the SpokePoolClient and other times query directly to the SpokePool. This prevents us from having being able to entirely encapsulate/abstract the SpokePool contract, which is what the MockSpokePoolClient tries to do for events.

For now I think the best way forward is to bump contracts-v2. After the recent updates it _should_ be ABI backwards compatible with current v2 deployments. Once we get that new package then we can deploy the contract in hre and interact with it as we normally would.